src/Component/Console/src/Helper/Table.js

```diff
@@ -0,0 +1,633 @@
+const InvalidArgumentException = Jymfony.Component.Console.Exception.InvalidArgumentException;
+const Helper = Jymfony.Component.Console.Helper.Helper;
+const OutputInterface = Jymfony.Component.Console.Output.OutputInterface;
+
+class Table {
+    /**
+     * @param {OutputInterface} output An OutputInterface instance
+     * @private
+     */
+    __construct(output) {
+        /**
+         * Table headers.
+         *
+         * @type {String[]}
```

Comment:
which is the difference between the scalar and the object? I'm asking this because in JS I can write `"this is a scalar".length` and it returns exactly the length of this string (and length is a property of `String`).

Reply:
from MDN:
String literals (denoted by double or single quotes) and strings returned from String calls in a non-constructor context (i.e., without using the new keyword) are primitive strings. JavaScript automatically converts primitives to String objects, so that it's possible to use String object methods for primitive strings. In contexts where a method is to be invoked on a primitive string or a property lookup occurs, JavaScript will automatically wrap the string primitive and call the method or perform the property lookup.

:+1: :)

Reply:
Is a subtle difference: the object contains the scalar value; you can use methods and query properties on a scalar value as such value is *implicitly* converted to a *temporary object* which is destroyed at the end of the call.
This means that you cannot define new members on a scalar value, but it is perfectly legal to define a new property on an object. Ex:

```js
const a = 'hello, world';
a.cust_prop = 'This is a custom property';     // implicit object conversion
console.log(a.cust_prop);    // prints undefined
```

and

```js
const a = new String('Hello, world!');
a.cust_prop = 'foobar';
console.log(a.cust_prop);    // prints "foobar"
```